### PR TITLE
Add SN-GAN

### PR DESCRIPTION
**Pull request by AlekseyKorshuk**
This implementation based on [this paper](https://arxiv.org/abs/1802.05957), [DCGAN](https://arxiv.org/abs/1511.06434) and [crypto-panks](https://github.com/vinyvince/cryptopunks-gan).

Model image size upscaled in comparison with other implementations of this GAN 🤗

Solved the problem of transparency with:
1. Using RGBA
2. Penalizing the model for transparency

Training script supports:

- [x]  Accelerate
- [x]  Push to hub
- [x]  Weights and Biases 
- [x]  Hub datasets
- [x]  All parameters from other training scripts from HugGAN

### CODE REVIEW

File: huggan/pytorch/sngan/README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+```bash
+accelerate launch train.py
+```
+
+This will create a local "images" directory, containing generated images over the course of the training.
+
+To train on another dataset available on the hub, simply do (for instance):
+
+```bash
+python train.py --dataset cifar-10
```
Comment (nateraw) [inline huggan/pytorch/sngan/README.md:31]:
```suggestion
python train.py --dataset cifar10
```

This issue is all around the repo, just a nit here :)

File: huggan/pytorch/sngan/train.py [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+            # Calculate FID metric
+            fid = calculate_fretchet(real_cpu, fake, model.to(accelerator.device))
+            logger.info(f"FID: {fid}")
+            if accelerator.is_local_main_process and args.wandb:
+                wandb.log({"FID": fid})
+
+    # Optionally push to hub
+    if accelerator.is_main_process and args.push_to_hub:
+        # os.makedirs(f"{args.output_dir}/{args.model_name}", exist_ok=True)
+        generator.push_to_hub(
```
Comment (nateraw) [inline huggan/pytorch/sngan/train.py:283]: If this doesn't work reliably, see:

Here:
https://github.com/huggingface/community-events/blob/15c92b6375d12febb9b8bde8b3abba70b429929d/huggan/pytorch/pix2pix/train.py#L115-L117

And Here:
https://github.com/huggingface/community-events/blob/15c92b6375d12febb9b8bde8b3abba70b429929d/huggan/pytorch/pix2pix/train.py#L292-L297

File: huggan/pytorch/sngan/train.py [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--dataset", type=str, default="mnist", help="Dataset to load from the HuggingFace hub.")
+    parser.add_argument(
+        "--num_workers", type=int, help="Number of data loading workers", default=0
+    )
+    parser.add_argument(
+        "--image_size",
+        type=int,
+        default=64,
```
Comment (nateraw) [inline huggan/pytorch/sngan/train.py:298]: Changing this doesn't work, eh? If so, don't be like us - just remove it 😉 haha. I think this flag should probably be removed across the repo if we don't have dynamic input sizes.

**Review by nateraw — COMMENTED** [review]
Wow thanks a lot for adding this!! 🚀 Will get a chance to run it a little later on tonight, but just wanted to drop a few comments when I first saw it.

File: huggan/pytorch/sngan/train.py (hunk 3 above)
Comment (AlekseyKorshuk) [reply huggan/pytorch/sngan/train.py:298]: Hahhaha. Best review comment ever :)

File: huggan/pytorch/sngan/train.py (hunk 2 above)
Comment (AlekseyKorshuk) [reply huggan/pytorch/sngan/train.py:283]: It works ok now. Just forgot to remove the comment

**Comment by AlekseyKorshuk** [comment]
Updated 👍
